store/sqlstore/post_store.go

```diff
@@ -889,6 +890,39 @@ func (s *SqlPostStore) Search(teamId string, userId string, params *model.Search
 			searchQuery = strings.Replace(searchQuery, "POST_FILTER", "", 1)
 		}
 
+		// handle after: before: on: filters
+		if len(params.AfterDate) > 1 || len(params.BeforeDate) > 1 || len(params.OnDate) > 1 {
```

Comment:
Would you be able to add some unit tests for these cases?

Reply:
i'll see what i can do

Reply:
@hmhealey I already added some unit tests for parsing these parameters etc. I don't see any existing tests for the Search function for me to add unit tests for my specific scenarios and don't really have time to do it for the entire Search function myself, let me know if I might have missed where it is implemented and I'll add test cases there.

Reply:
@dmitrysamuylovpharo The tests for the store code is in store/storetest/post_store.go including the search function. They're written a bit differently for the store package so that they can be properly mocked when testing. There's also tests for post searching in api4/post_test.go

Reply:
@hmhealey I see it, thanks for pointing it out, I don't see any flag based search tests in there but let me give it a try